src/library-authoring/component-info/ComponentInfo.tsx

```diff
@@ -146,6 +147,19 @@ const ComponentInfo = () => {
   const canPublish = (new Date(componentMetadata?.modified ?? 0)) > (new Date(componentMetadata?.lastPublished ?? 0));
   const { showToast } = React.useContext(ToastContext);
 
+  const { client, indexName } = useContentSearchConnection();
+  const {
+    hits: componentHit,
+  } = useContentSearchResults({
+    client,
+    indexName,
+    searchKeywords: '',
+    extraFilter: [`usage_key IN ["${usageKey}"]`],
+    limit: 1,
+    enabled: true,
+    skipBlockTypeFetch: true,
+  });
+
```

Comment:
What do you think about creating a `useComponentUnits` hook for this?
Also, I believe that if you move the fetch call to the `ComponentDeleter`, you will not need to replicate this on the `LibraryUntiBlocks`.

Reply:
I added the `useComponentsFromSearchIndex` in https://github.com/openedx/frontend-app-authoring/pull/1906/commits/917421dd36c3c3636d0395d8c2e30f3a154c2c7c.  

About moving the fetch all to the `ComponentDeleter`, I think I would add more calls, in the case of the card component, that information is already available from the search index, and if we add it to the `ComponentDeleter`, it would be called again

Reply:
@ChrisChV Can we move the call inside of `ComponentMenu` while still accepting `units` prop and only fetch if the prop is not passed?

Reply:
Or another solution would be to put it inside the `ComponentDeleter` and change to render it conditionally, like `{isConfirmingDelete && <ComponentDeleter usageKey={usageKey} isConfirmingDelete={isConfirmingDelete} cancelDelete={cancelDelete} />}`.
That way, the component is only rendered when the dialog opens, saving the extra calls.

Reply:
Refactored here: https://github.com/openedx/frontend-app-authoring/pull/1906/commits/6291f67dbe69d4776127c138b0b1d9f22bb9c170